Cypress: Fix OLM test flakes

dtaylor113 opened the pull request:
- consolidated non-independant `it` tests, into single test
- Added check for loading indicators to not exist
- Add check for url to contain '~new`

dtaylor113 commented:
/retest

spadgett commented:
Based on the video I'm looking at for at least one failure, the form is not loading. I think this might be a legitimate bug rather than simply a test flake.

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/batch/pull-ci-openshift-console-master-e2e-gcp-console/1372889154475528192/artifacts/e2e-gcp-console/test/artifacts/gui_test_screenshots/cypress/videos/operator-install-single-namespace.spec.ts.mp4

rhamilto commented:
> * consolidated non-independant `it` tests, into single test

we should also do this for https://github.com/openshift/console/blob/master/frontend/packages/operator-lifecycle-manager/integration-tests-cypress/tests/operator-install-global.spec.ts so they're consistent?

dtaylor113 commented:
> Based on the video I'm looking at for at least one failure, the form is not loading.

Might be related to this: https://stackoverflow.com/questions/51327713/why-cypress-not-open-modal-page
I'll check if our modals are using annimations/transitions

dtaylor113 commented:
@rhamilto & @spadgett, I noticed from the video and running it locally that after clicking on the 'Create ...' the loading dots appear for a split second which might be causing the issue.  Added a check to wait for loading dots to not exist.

Also updated `operator-install-global.spec` to reduce to independant `it` tests.

rhamilto commented on frontend/packages/operator-lifecycle-manager/integration-tests-cypress/tests/operator-install-single-namespace.spec.ts:
```diff
@@ -58,42 +58,31 @@ describe(`Interacting with a single namespace install mode Operator (${operatorN
     cy.byTestOperatorRow(operatorRow, { timeout: 60000 }).should('exist');
-  });
-
-  it(`displays details about ${operatorName} ClusterServiceVersion on the "Details" tab`, () => {
```
In the global test, you migrated these to a `cy.log`, which I like.  Do the same here?

dtaylor113 replied:
Yes, but in most cases the `it` was followed immediately by a `cy.log`, ie:
```
it(`displays empty message on the ${operatorName} ClusterServiceVersion "All Instances" tab`, () => {
    cy.log('navigate to the "All Instances" tab');
```
In these cases, I tended to use the `cy.log`, because it was concise, whereas the `it` I feel is verbose, and a new overall `it` had been created.

rhamilto replied:
If that's the case, you don't need `operatorInstance` as a `const` since it is only used once.

dtaylor113 replied:
fixed, ty

dtaylor113 commented:
Hmm...another type of failure/flake?  cc @rhamilto -thanks
**Global**
![image](https://user-images.githubusercontent.com/12733153/112152223-70f0ba80-8bb8-11eb-8696-59367041ab07.png)
**Single**
![image](https://user-images.githubusercontent.com/12733153/112152256-7b12b900-8bb8-11eb-9506-eaa5eb24de10.png)

dtaylor113 commented:
/retest

dtaylor113 commented:
/test all
passed: 1

dtaylor113 commented:
/test all
passed: 2

dtaylor113 commented:
/retest

dtaylor113 commented:
/retest

dtaylor113 commented:
/test ci/prow/e2e-gcp-console

dtaylor113 commented:
/test e2e-gcp-console

dtaylor113 commented:
@rhamilto & @spadgett, still seeing [OLM flakes](https://search.ci.openshift.org/?search=AssertionError&maxAge=168h&context=1&type=junit&name=e2e-gcp-console&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job).   I'd like to merge this PR to see if it cuts down on the flakes.

rhamilto commented:
Ok by me.  @spadgett?

spadgett commented:
> @rhamilto & @spadgett, still seeing [OLM flakes](https://search.ci.openshift.org/?search=AssertionError&maxAge=168h&context=1&type=junit&name=e2e-gcp-console&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job). I'd like to merge this PR to see if it cuts down on the flakes. Consolidating `it` tests will at least do a 'retry' from a correct start state.

Hey, @dtaylor113. My concern is: I don't think they are flakes. It looks like a legitimate bug to me. If we improve the retries so that the tests stop failing, that just covers up the problem instead of fixing it, and we lose the only way we have to reproduce currently (CI). I think we need to debug what's happening here.

If we're sure it's a test flake, OK. But I'm not convinced that's the case.

dtaylor113 commented:
> If we're sure it's a test flake, OK. But I'm not convinced that's the case.

I've never been able to reproduce in local cypress runs, nor have I've seen this happen in the actual concole.

dtaylor113 commented:
closing in favor of addressing [BZ 1976939](https://bugzilla.redhat.com/show_bug.cgi?id=1976939) and [BZ 1966499](https://bugzilla.redhat.com/show_bug.cgi?id=1966499) which deal with rewriting the OLM tests to use Red Hat Operators instead of Community Operators.
